lumigator/backend/backend/services/jobs.py

```diff
@@ -309,108 +305,133 @@ def _validate_config(self, job_type: str, config_template: str, config_params: d
         else:
             loguru.logger.info(f"Validation for job type {job_type} not yet supported.")
 
-    def _get_job_params(self, job_type: JobType, record, request: BaseModel) -> dict:
-        # get dataset S3 path from UUID
-        dataset_s3_path = self._dataset_service.get_dataset_s3_path(request.dataset)
+    # The end result should be that InferenceJobConfig is actually JobInferenceConfig
+    # (resp. Eval)
+    # For the moment, something will convert one into the other, and we'll decide where
+    # to put this. The jobs should ideally have no dependency towards the backend.
 
-        # provide a reasonable system prompt for services where none was specified
-        if (
-            job_type in [JobType.EVALUATION, JobType.INFERENCE]
-            and request.system_prompt is None
-            and not request.model.startswith("hf://")
-        ):
-            request.system_prompt = settings.DEFAULT_SUMMARIZER_PROMPT
-
-        # Base job parameters (used for eval-lite, evaluation and inference).
-        job_params = {
-            "dataset_path": dataset_s3_path,
-            "job_id": record.id,
-            "job_name": request.name,
-            "max_samples": request.max_samples,
-            "model_uri": request.model,
-            "storage_path": self.storage_path,
-        }
+    def generate_inference_job_config(
+        self, request: JobCreate, record_id: UUID, dataset_path: str, storage_path: str
+    ):
+        job_config = InferenceJobConfig(
+            name=f"{request.name}/{record_id}",
+            dataset=IDatasetConfig(path=dataset_path),
+            job=InferJobConfig(
+                max_samples=request.max_samples,
+                storage_path=storage_path,
+                output_field=request.job_config.output_field,
+            ),
+        )
+        # Maybe use just the protocol to decide?
+        match request.job_config.model:
+            case "oai://gpt-4o-mini" | "oai://gpt-4o":
```

Comment:
I think eventually we will want to elevate this logic maybe into a separate class (something like InferenceManager) as we support more complex prompts and models. However, I think for the time being this refactor is moving in the right direction and totally makes sense 👍

Reply:
As commented, my idea could be that the InferJobConfig would be passed roughly transparently to the jobs (and then we can definitely use a base class on top of WhateverJobConfig to hold the job-backend interaction details like storage_path or dataset_path, since here we aren't concerned about the OpenAPI translation).

I'll resolve then this issue, please unresolve if there's any other aspect you'd like to discuss.